Sources/SpeziBluetooth/CoreBluetooth/BluetoothPeripheral.swift

```diff
@@ -319,17 +336,22 @@ public actor BluetoothPeripheral: BluetoothActor { // swiftlint:disable:this typ
             self.invalidateServices(Set(services.map { $0.uuid }))
         }
 
+        connectAccess.cancelAll()
         characteristicAccesses.cancelAll()
         writeWithoutResponseAccess.cancelAll()
         rssiAccess.cancelAll()
 
+        if let connectContinuation {
+            self.connectContinuation = nil
+            connectContinuation.resume(throwing: error ?? CancellationError())
+        }
         if let writeWithoutResponseContinuation {
             self.writeWithoutResponseContinuation = nil
             writeWithoutResponseContinuation.resume()
         }
         if let rssiContinuation {
             self.rssiContinuation = nil
-            rssiContinuation.resume(throwing: CancellationError())
+            rssiContinuation.resume(throwing: error ?? CancellationError())
```

Comment:
I would not throw the connection error into the rssi continuation. I would propose to just return the CancellationError in every case. Or what do you think?